packages/core/data/oot/world/shadow_temple.yml

```diff
@@ -25,17 +25,18 @@
 "Shadow Temple Main":
   dungeon: Shadow
   exits:
-    "Shadow Temple Open": "small_keys_shadow(1) && has_explosives"
+    "Shadow Temple Open": "cond(climb_anywhere, small_keys_shadow(4), small_keys_shadow(1)) && has_explosives"
     "Shadow Temple Scythe Silver Rupees": "true"
   locations:
-    "Shadow Temple Compass": "soul_redead_gibdo && (has_weapon || can_use_sticks || can_use_din)"
+    "Shadow Temple Compass": "soul_redead_gibdo && (has_weapon || can_use_sticks || can_use_din || can_hammer)"
 "Shadow Temple Scythe Silver Rupees":
   dungeon: Shadow
   exits:
     "Shadow Temple Main": "true"
+    "Shadow Temple Boat": "(climb_anywhere && trick(OOT_SHADOW_BOAT_EARLY)) || event(SHADOW_SHORTCUT)"
```

Comment:
how do you get past the block with climb anywhere?

Reply:
you just climb up the cage. you pop up on top.